# "Analyzing International Debt Statistics"

> "Foreign debt is money borrowed by a government, corporation or private household from another country's government or private lenders"


- toc: true
- badges: true
- comments: true
- categories: [SQL, Data visualization, data mining,python]
- image: images/sri_lankas_dept.png
- author: Alphonse Brandon

## Background

<p>Foreign debt, also known as external debt, has been rising steadily in recent decades, with unwelcome side-effects in some borrowing countries. These include slower economic growth, particularly in low-income countries, as well as crippling debt crises, financial market turmoil, and even secondary effects such as a rise in human-rights abuses.</p>

## Goal

<p>In this notebook, we are going to analyze international debt data collected by The World Bank. The dataset contains information about the amount of debt (in USD) owed by developing countries across several categories. We are going to find the answers to questions like:</p>
<ol>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
<ol>


![](/images/dept.jpg "Image")

I will be using MySQL database for this project and using a dataset gotten from the World Bank



Let's start by importing the necessary packages

In [16]:
import mysql.connector
import sqlalchemy
import pandas

Now let's make a connection to the database

In [17]:
sqlalchemy.create_engine("mysql://root:@127.0.0.1")


Engine(mysql://root:***@127.0.0.1)

Loading the sql magic extension to use sql directly in jupyter notebook

In [18]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


Connecting to my database table

In [19]:
%%sql 
mysql://root:@localhost/debt_statistics

## 1. Let's display The first five rows to know the content of the dataset and the different data points

In [20]:
%%sql

SELECT * FROM international_debt
LIMIT 5

 * mysql://root:***@localhost/debt_statistics
5 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001'))]

### 2. Let's get to know the number of distint countries present in our dataset

From our dataet there is repetition in the country name accross several rows as the country is most likely to have debt in more than on debt indicators

To helt us filter this, let's use the count distinc sql function to extract unique countries present in the database

In [22]:
%%sql

SELECT
    COUNT(DISTINCT country_name) AS total_distinct_countries
FROM international_debt


 * mysql://root:***@localhost/debt_statistics
1 rows affected.


[(124,)]

As seen from our query above, there are 124 countries contained in this dataset

### 3. Now Let's find the number of distinct debt indicators

From our table, there is a column name called indicator name that highlights the purpose of taking debt.

There is also a column close to it called indicator_code which symbolizes the cathegory of the debt

This will help us find out about the areas where countries are amostly indebted in.

In [26]:
%%sql
SELECT DISTINCT indicator_code AS distinct_debt_indicators
FROM international_debt
ORDER BY distinct_debt_indicators

LIMIT 5

 * mysql://root:***@localhost/debt_statistics
5 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',)]

### 4. Summing the amount of debt owed by all countries

This will give us a glims of how the economy of the world is doing, so let's find the total debt owed by counties in USD